app/models/user.rb

```diff
@@ -262,6 +262,11 @@ def subordinates
     User.active.where(agency_id: self.agency_id).order(:first_name, :last_name)
   end
 
+  def coc_codes_for_consent
+    return coc_codes if coc_codes.present?
+    GrdaWarehouse::Hud::ProjectCoc.distinct.order(:CoCCode).pluck(:CoCCode)
```

Comment:
So, if a user doesn't have coc_codes, return all the ones in the system currently? I'm concerned about the impact of adding a CoC to a warehouse as a result...

Reply:
This is used to populate the list of available CoCs.  If the user doesn't have any assigned, they can see the full list and pick appropriately when uploading a consent form.